src/settingsmanager.cpp

```diff
@@ -176,6 +176,8 @@ void SettingsManager::initializeSettingsLibrary()
     settingsLibrary.insert("slideshowtimer", {5, {}});
     settingsLibrary.insert("afterdelete", {2, {}});
     settingsLibrary.insert("askdelete", {true, {}});
+    settingsLibrary.insert("showhiddenfiles", {false, {}});
+    settingsLibrary.insert("allowmimecontentdetection", {true, {}});
```

Comment:
Defaults to true to preserve existing behavior. If a file extension isn't recognized, or sometimes even if it is but could refer to multiple mime types, QMimeDatabase will inspect the file contents. This of course creates additional I/O, especially noticeable with large network shares, and toggling this setting off can help massively.